src/neptune/internal/operation_processors/async_operation_processor.py

```diff
@@ -107,6 +129,20 @@ def wait(self):
         if not self._consumer.is_running():
             raise NeptuneSynchronizationAlreadyStoppedException()
 
+    def _check_lag(self):
+        if not self._lag_exceeded and self._last_ack and monotonic() - self._last_ack > self._async_lag_threshold:
+            with self._lock:
+                if not self._lag_exceeded:
+                    self._async_no_progress_callback()
+                    self._lag_exceeded = True
+
+    def _check_no_progress(self):
+        if self._should_call_no_progress_callback:
```

Comment:
Could we rewrite it slightly so that it's less indented?

```
if not self._should_call_no_progress_callback:
    return
with self._lock:
    if self._should_call_no_progress_callback:
        self._async_no_progress_callback()
        self._should_call_no_progress_callback = False
```

Reply:
Also, is it necessary to check this condition twice?
`if self._should_call_no_progress_callback`

Reply:
This is common pattern as lock is heavy operation: https://en.wikipedia.org/wiki/Double-checked_locking

Reply:
Done